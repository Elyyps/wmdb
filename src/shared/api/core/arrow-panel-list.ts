import PlaceholderImage from "@assets/img02.png";
import SPEL from "@assets/icons/spel.svg";
import ETEN from "@assets/icons/eten.svg";
import ROOMS from "@assets/icons/rooms.svg";
import { IArrowPanelListComponentProps } from "../../core/arrow-panel-list/arrow-panel-list.component";

export const GenerateDummyArrowPanelListData = () => ({
  title: "Spel & Ontspanning",
  icon: ETEN,
  items: [
    {
      title: "Wat is Lorem Ipsum?",
      link: "/"
    },
    {
      title: "Hoe lang zijn mijn aanbiedingen of vacatures zichtbaar op indebuurt?",
      link: "/"
    },
    {
      title: "Hoe kan ik mijn (wekelijkse) aanbieding of vacature aanleveren?",
      link: "/"
    },
    {
      title: "Hoe lang staat mijn artikel op de homepagina, waar komt het vandaan?",
      link: "/"
    },
    {
      title: "Waar kan ik het vinden?",
      link: "/"
    },
    {
      title: "Waar kan ik het vinden?",
      link: "/"
    }
  ],
  linkMore: {
    title: "Alle veelgestelde vragen over musea",
    link: "/"
  }
});
