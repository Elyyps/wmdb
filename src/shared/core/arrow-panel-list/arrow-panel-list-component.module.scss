.arrow-panel-list {
  &__posts {
    @media (min-width: $breakpoint-small) {
      margin-bottom: -24px;
    }
    @media (min-width: $breakpoint-medium) {
      margin-bottom: -40px;
    }
    > div {
      @media (min-width: $breakpoint-small) {
        padding-bottom: 24px;
      }
      @media (min-width: $breakpoint-medium) {
        padding-bottom: 40px;
      }
    }
    @media (max-width: $breakpoint-xsmall-max) {
      > div:not(:first-child) {
        display: none;
      }
    }
  }

  &__head {
    display: flex;
    align-items: center;
    margin-bottom: 24px;
    @media (max-width: $breakpoint-xsmall-max) {
      margin-bottom: 16px;
    }
    > h3 {
      margin: 0;
    }
    &__icon {
      margin-right: 16px;
      @media (max-width: $breakpoint-xsmall-max) {
        margin-right: 8px;
        > * {
          width: 24px !important;
        }
      }
    }
  }

  &__item {
    margin-bottom: 1px;
  }
}
