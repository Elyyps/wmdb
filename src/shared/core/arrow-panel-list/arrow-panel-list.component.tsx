import * as React from "react";
import styles from "./arrow-panel-list-component.module.scss";
import { IconComponent } from "../icon";
import { ArrowPanel, IArrowPanelProps } from "../arrow-panel/arrow-panel.component";
import { SeeMorePanel } from "../see-more-panel";

export interface IArrowPanelListComponentProps {
  icon?: string;
  items?: IArrowPanelProps[];
  linkMore?: IArrowPanelProps;
  title: string;
}

const ArrowPanelListComponent = (props: IArrowPanelListComponentProps) => {
  const { items, title, icon, linkMore } = props;

  return (
    <div className={styles["arrow-panel-list"]}>
      <div className={styles["arrow-panel-list__head"]}>
        {icon && (
          <div className={styles["arrow-panel-list__head__icon"]}>
            <IconComponent icon={icon} size="34px" />
          </div>
        )}
        <h3>{title}</h3>
      </div>
      {items &&
        items.map((item: IArrowPanelProps, key: number) => (
          <div key={key} className={styles["arrow-panel-list__item"]}>
            <ArrowPanel title={item.title} link={item.link} />
          </div>
        ))}
      {linkMore && (
        <div className={styles["arrow-panel-list__item"]}>
          <SeeMorePanel title={linkMore.title} link={linkMore.link} />
        </div>
      )}
    </div>
  );
};
export { ArrowPanelListComponent };
