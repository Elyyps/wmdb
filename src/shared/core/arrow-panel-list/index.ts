export * from "./arrow-panel-list.component";
