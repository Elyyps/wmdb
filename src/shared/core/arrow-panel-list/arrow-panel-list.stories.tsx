import * as React from "react";

import { storiesOf } from "@storybook/react";
import { withA11y } from "@storybook/addon-a11y";
import { withProvider } from "@app/util";
import { ArrowPanelListComponent } from "./arrow-panel-list.component";

storiesOf("ArrowPanelList", module)
  .addDecorator(withA11y)
  .addDecorator(withProvider)
  .add("Basic implementation", () => <ArrowPanelListComponent title="title" />);
