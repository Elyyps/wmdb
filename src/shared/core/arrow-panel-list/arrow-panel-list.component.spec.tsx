import React from "react";
import { shallow } from "enzyme";
import { ArrowPanelListComponent } from "./arrow-panel-list.component";

describe("[ArrowPanelList]", () => {
  it("should render component without crashing", () => {
    const renderedComponent = shallow(<ArrowPanelListComponent title="Title" />);
    expect(renderedComponent).toMatchSnapshot();
  });
});
