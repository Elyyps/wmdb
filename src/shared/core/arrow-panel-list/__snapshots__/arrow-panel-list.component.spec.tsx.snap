// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[ArrowPanelList] should render component without crashing 1`] = `
<div
  className="arrow-panel-list"
  style={Object {}}
>
  <div
    className="arrow-panel-list__head"
    style={Object {}}
  >
    <h3>
      Title
    </h3>
  </div>
</div>
`;
